src/webgpu/api/validation/query_set/create.spec.ts

```diff
@@ -0,0 +1,178 @@
+export const description = `
+Tests for validation in createQuerySet.
+`;
+
+import { params, poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+const kMaxQueryCount = 8192;
+
+// Test with these query counts.
+const kQueryCounts: number[] = [
+  // valid
+  0,
+  1,
+  128,
+  kMaxQueryCount,
+  // invalid
+  kMaxQueryCount + 1,
+];
+
+g.test('occlusion')
+  .desc(
+    `
+Tests that create query set with the type of occlusion.
+- control cases:
+  - count <= kMaxQueryCount
+  - pipelineStatistics is undefined
+- invalid cases:
+  - count > kMaxQueryCount
+  - pipelineStatistics is set with pipeline statistic names
+  `
+  )
+  .params(
+    params()
+      .combine(poptions('count', kQueryCounts))
+      .combine(
```

Comment:
It's not necessary to test cases that are invalid for multiple reasons (count is too large AND pipelineStatistics is invalid). You can use something like this: https://github.com/gpuweb/cts/pull/425#discussion_r554262676